File: module_5/Exercises Review/exercises-review/src/app/model/Product.ts
export class Product {
  private _name: string;
  private _price: number;
  private _productDate: string;
  private _quantity: number;
  private _description: string;


  constructor(name: string, price: number, productDate: string, quantity: number, description: string) {
    this._name = name;
    this._price = price;
    this._productDate = productDate;
    this._quantity = quantity;
    this._description = description;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get price(): number {
    return this._price;
  }

  set price(value: number) {
    this._price = value;
  }

  get productDate(): string {
    return this._productDate;
  }

  set productDate(value: string) {
    this._productDate = value;
  }

  get quantity(): number {
    return this._quantity;
  }

  set quantity(value: number) {
    this._quantity = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }
}

File: module_4/test_module_4/src/main/java/com/test_module_4/model/request/Product.java
package com.test_module_4.model.request;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "product")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private Integer productId;
    @Column(name = "product_name", length = 45)
    private String productName;
    @Column(name = "price", length = 45)
    private String price;
    @Column(name = "description", length = 45)
    private String description;

    @ManyToOne
    @JoinColumn(name = "product_type_id")
    private ProductType productType;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<Request> requestList;


}
